### ISSUE TITLE
MSVC: allow nested quotes

### ISSUE THREAD

**Issue by justinpryzby** [ccache/ccache#1021]
1. Compiling postgresql with cirrus CI.  I don't have a local MSVC...
2. A small number of files fail to compile due to being called with things like: /D "HOST_TUPLE=\\"i686-pc-win32vc\\""
```
[04:51:47.170] c:\cirrus\src\test\regress\pg_regress.c(55,26): error C2017: illegal escape sequence [c:\cirrus\pg_regress_ecpg.vcxproj]
[04:51:47.170] c:\cirrus\src\test\regress\pg_regress.c(55,26): error C2065: 'i686': undeclared identifier [c:\cirrus\pg_regress_ecpg.vcxproj]
[04:51:47.170] c:\cirrus\src\test\regress\pg_regress.c(55,26): error C2065: 'pc': undeclared identifier [c:\cirrus\pg_regress_ecpg.vcxproj]
[04:51:47.170] c:\cirrus\src\test\regress\pg_regress.c(55,36): error C2065: 'win32vc': undeclared identifier [c:\cirrus\pg_regress_ecpg.vcxproj]
[04:51:47.170] c:\cirrus\src\test\regress\pg_regress.c(55,36): error C2099: initializer is not a constant [c:\cirrus\pg_regress_ecpg.vcxproj]
```

### Actual behavior ###
String quoting differed when ccache was used.

### Expected behavior ###
The same behavior with ccache as without.

### Environment ###
4.6 / GIT HEAD + some patches

> [04:46:36.267] -- Building for: Visual Studio 16 2019
> [04:46:36.293] -- Selecting Windows SDK version 10.0.20348.0 to target Windows 10.0.17763.
> [04:46:41.756] -- The C compiler identification is MSVC 19.29.30139.0

**Comment by justinpryzby** [comment]
You can see the compilation here.  https://cirrus-ci.com/task/6093519424061440
I was able to work around the issue with this patch.
https://github.com/justinpryzby/ccache/commit/e0e693c677a4086a212d6a264954c13c6ae9529b
Maybe @orgads wants to know about this.

**Comment by jrosdahl** [comment]
OK, so apparently the MSVC RSP file format is not simply GCC's format without parsing backslashes but something else. Therefore we need to (a) figure out what the actual format of RSP files is, (b) describe the expected behavior of the "ignore_backslash" mode of `Args::from_atfile"` in corresponding unit tests (or potentially write a completely different implementation for the MSVC case), and (c) implement the expected behavior.

I don't have bandwidth to do this myself, so any help is welcome.

**Comment by justinpryzby** [comment]
I think it *is* the RSP format used by MSVC, or at least it's intended to be.  See libiberty/argv.c build_argv().
If it'd work for you, I could polish what I've written and write a unit test to handle at least the cases that I know it needs to handle.

**Comment by jrosdahl** [comment]
@justinpryzby wrote:
> If it'd work for you, I could polish what I've written and write a unit test to handle at least the cases that I know it needs to handle.

Sure, that would be great.

**Comment by Predelnik** [comment]
Wasn't this issue fixed with #1071?

**Comment by justinpryzby** [comment]
Yes, confirmed.  ccache 4.6.1 can now compile postgresql.
Currently, it does not improve the compilation speed - but maybe the depend mode patch will help @orgads
